dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -958,18 +958,40 @@ private void processLink(Reader source, Listener listener) throws IOException
                             new ArrayList<>(functionBuffer
                                     .subList(1, functionBuffer.size() - 2))), ResourceType.MAILTO);
                     reference.setTyped(true);
-                } else {
-                    reference = new ResourceReference(getStringRepresentation(
-                            new ArrayList<>(functionBuffer.subList(0, functionBuffer.size() - 1))), ResourceType.URL);
-                    reference.setTyped(false);
+                    listener.beginLink(reference, false, Listener.EMPTY_PARAMETERS);
+                    processWordsFromReader((char) c, source, listener, TAG_DOUBLE_CLOSING_SQUARE_BRACKETS);
+                    listener.endLink(reference, false, Listener.EMPTY_PARAMETERS);
+                    break;
+                }
+                else {
+                    if (c != '{'){
```

Comment:
This does not follow XWiki codestyle (missing white space).

You should configure your IDE formatter with the XWiki setup, see http://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HConfiguringyourIDEtousetheXWikicodestyle.

Reply:
Cool. 👍